File: python/알고리즘/baek/브루트포스/14465_소가길을건너간이유5.py
N, K, B = map(int, input().split())

trafficlight = [1] * (N+1)

# 고장난 애들 체크
for _ in range(B):
    broken = int(input())
    trafficlight[broken] = 0

# 0, 0, 0, 1, 1, 0, 1, 1, 1, 0, 0
# 5 번을 고치면 댐.

need = 0
for i in range(1, K+1):
    if trafficlight[i] == 0:
        need += 1

result = need

for i in range(2, N-K+2):
    # 2,3,4,5,6,7
    # ...
    # 5,6,7,8,9,10
    if trafficlight[i-1] == 0:
        need -= 1
    
    if trafficlight[i+K-1] == 0:
        need += 1
    
    result = min(result, need)

print(result)

File: python/알고리즘/baek/브루트포스/14465_소가길을건너간이유5.c
#include <stdio.h>
#define min(a,b)   (((a) < (b)) ? (a) : (b))

int N, K, B;
int TrafficLight[100000+10] = {1,};

void InputData(void){
    scanf("%d %d %d %d", &N, &K, &B);
    int Empty;
    for (int i=0; i<B ; i++){
        scanf("%d", &Empty);
        TrafficLight[Empty] = 0;
    }
}


int main(void) {
    InputData();

    int need = 0;
    for (int i = 1; i < K+1; i++) {
        if (TrafficLight[i] == 0) {
            need++;
        }
    }
    int result = need;
    for (int i = 2; i < N - K + 2; i++) {
        if (TrafficLight[i-1] == 0) {
            need--;
        }
        if (TrafficLight[i+K-1] == 0) {
            need++;
        }
        
        result = min(result, need);
    }

    printf("%d", result);
    return 0;
}